rpcs3/util/atomic.cpp

```diff
@@ -1181,15 +1322,29 @@ atomic_wait_engine::notify_all(const void* data, u32 size, __m128i mask, __m128i
 		// Proceed with remaining bits using "normal" blocking waiting
 		for (u64 bits = copy; bits; bits &= bits - 1)
 		{
-			cond_get(lock_ids[std::countr_zero(bits)])->alert_native();
+			const u32 id = std::countr_zero(bits);
+
+			const auto cond_id = lock_id2[id] ? lock_id2[id] : lock_ids[id];
+
+			cond_get(cond_id)->alert_native();
 
 			s_tls_notify_cb(data, ++progress);
 		}
 
 		// Cleanup locked notifiers
 		for (u64 bits = lock; bits; bits &= bits - 1)
 		{
-			cond_free(lock_ids[std::countr_zero(bits)]);
+			const u32 id = std::countr_zero(bits);
+
+			if (u32 cond_id = lock_id2[id])
+			{
+				cond_free(cond_id);
+			}
+
+			if (u32 cond_id = lock_ids[id])
```

Comment:
i think this should be ```else if``` to not call ```cond_free``` twice

Reply:
it's two different cond_id and they both should be deallocated

Reply:
Oh, my bad. Just not that familiar with such syntax.